src/common/commands_parser.rs

```diff
@@ -265,6 +265,62 @@ impl PcrArgs {
     }
 }
 
+/// The arguments used by `sign-eif` command
+#[derive(Debug, Clone)]
+pub struct SignEifArgs {
+    /// Path to the EIF file needed for signing
+    pub eif_path: String,
+    /// The path to the signing certificate for signed enclaves.
+    pub signing_certificate: Option<String>,
+    /// The path to the private key for signed enclaves.
+    pub private_key: Option<String>,
+    /// ID of the KMS key for signed enclaves.
+    pub kms_key_id: Option<String>,
+    /// Region of the KMS key for signed enclaves.
+    pub kms_key_region: Option<String>,
+}
+
+impl SignEifArgs {
+    /// Construct a new `SignEifArgs` instance from the given command-line arguments.
+    pub fn new_with(args: &ArgMatches) -> NitroCliResult<Self> {
+        let signing_certificate = parse_signing_certificate(args);
+        let private_key = parse_private_key(args);
+        let kms_key_id = parse_kms_key_id(args);
+        let kms_key_region = parse_kms_key_region(args);
+
+        validate_signing_credentials(&signing_certificate, &private_key, &kms_key_id)?;
```

Comment:
a bit weird that `kms_key_region` is not validated

Reply:
no, I mean potentially we could have region and private key at the same time

Reply:
Which is why you could just pass the ARN of the KMS key in as the `--private-key` argument... :)